matRad_calcParticleDose.m

```diff
@@ -85,22 +85,35 @@
 if strcmp(pln.bioOptimization,'effect') || strcmp(pln.bioOptimization,'RBExD') ... 
         && strcmp(pln.radiationMode,'carbon')
     fprintf('matRad: loading biological base data... ');
-    mTissueClass = zeros(size(V,1),1);
+    vTissueIndex = zeros(size(V,1),1);
+    
+    %set overlap priorities
+    cst  = matRad_setOverlapPriorities(cst);
+    
     for i = 1:size(cst,1)
         % find indices of structures related to V
         [~, row] = ismember(cst{i,4},V,'rows');  
-        if ~isempty(cst{i,5}) && isfield(cst{i,5},'TissueClass')
-            mTissueClass(row) = cst{i,5}.TissueClass;
+        if ~isempty(cst{i,5}) && isfield(cst{i,5},'alphaX') && isfield(cst{i,5},'betaX')
+            
+            IdxTissue = find(ismember(machine.data(1).alphaX,cst{i,5}.alphaX) & ...
+                                     ismember(machine.data(1).betaX,cst{i,5}.betaX));
+                                 
+            % check consitency of biological baseData and cst settings
+            if ~isempty(IdxTissue)
+                vTissueIndex(row) = IdxTissue;
+            else
+                error('biological base data and cst inconsistent\n');
+            end
         else
-            mTissueClass(row) = 1;
+            vTissueIndex(row) = 1;
             fprintf(['matRad: tissue type of ' cst{i,2} ' was set to 1 \n']);
         end
         
-        % check consitency of biological baseData and cst settings
-        baseDataAlphaBetaRatios = reshape([machine.data(:).alphaBetaRatio],numel(machine.data(1).alphaBetaRatio),size(machine.data,2));
-        if norm(baseDataAlphaBetaRatios(cst{i,5}.TissueClass,:) - cst{i,5}.alphaX/cst{i,5}.betaX)>0
-            error('biological base data and cst inconsistent\n');
-        end
+%         % check consitency of biological baseData and cst settings
+%         baseDataAlphaBetaRatios = reshape([machine.data(:).alphaBetaRatio],numel(machine.data(1).alphaBetaRatio),size(machine.data,2));
+%         if norm(baseDataAlphaBetaRatios(cst{i,5}.TissueClass,:) - cst{i,5}.alphaX/cst{i,5}.betaX)>0
+%             error('biological base data and cst inconsistent\n');
+%         end
```

Comment:
Should the check not be directly on alphaX and betaX? Otherwise we could have some issues with different absolute values of alphaX and betaX that have the same ratio?!